### PR TITLE
Refactoring class to function.

### PR DESCRIPTION
**Pull request by franclobo**
### Project requirements.

- Refactor the Calculator component (components/Calculator.js) from a class-based component to a functional component with hooks.

### CODE REVIEW

File: src/components/calculator.js [hunk 1]
```diff
@@ -2,22 +2,15 @@ import React, { useState } from 'react';
...
-    this.clickEvent = this.clickEvent.bind(this);
-  }
-
-  clickEvent() {
-    const { clickEvent, btnName } = this.props;
+function Buttons(props) {
+  const { className, btnName } = props;
+
+  const clickEvent = (props) => {
+    const { clickEvent, btnName } = props;
```
Comment (codepantha) [inline src/components/calculator.js]: - [ ] The app buttons don't work anymore. As a result, I'm unable to make any mathematical computation with your calculator app. Here's an image of the console error 👇 
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUESTED CHANGES ♻️ 

## Hi @franclobo 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### Highlights ⭐ 
✔️ Descriptive PR
✔️ Linter checks

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [x] In your `calculator.js` file inside `./components` directory, you have a `<Button />` and `<Display />` component but they are defined inside the same file. This approach works but may not be the best setup for reusability, plus it makes your file avoidably long.
Please consider putting them in their own files and exporting them wherever you may need them 👍 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/calculator.js (hunk 1 above)
Comment (franclobo) [reply src/components/calculator.js]: Fixed.

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Approved 🟢🟢🟢 

Hi @franclobo,

YesYouAreGIF

Good job working on the changes asked by the previous reviewer 👍 
Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! 🎉 

## Optional Changes ♻️♻️♻️ 
- [x] I noticed that you're missing out on the operation for modulus(%). The previous reviewer did not spot that. Kindly add it as it's a major requirement you're supposed to do. The operation for the modulus(%) operator is in `operate.js` but you did not add a button for it in your calculator component. Please add it and submit and check that it works before you submit it for a review in order for your project to be approved 👍 Please fix that

image


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by AmaduKamara** [comment]
Hi @franclobo,
I previously asked for changes on the project relating to the modulus operator. But I made a mistake by approving your project on the review dashboard. It's not reversible and hence I changed the comment to approval. 

- [x] Kindly work on the change I marked as optional. It's supposed to be implemented and it will affect you on the next project if you don't fix it. Please take your time and fix it 👍

**Comment by AmaduKamara** [comment]
Please, you don't have to send for a review anymore

**Comment by AmaduKamara** [comment]
Make the changes and merge your project then submit on your dashboard because it's already signaling to Submit right now. 👍

**Pull request merged**